nixos/modules/services/mail/postfix.nix

```diff
@@ -201,6 +201,23 @@ in
         default = true;
         description = "Whether to enable smtp in master.cf.";
       };
+      
+      enableSubmission = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable smtp submission";
+      };
+
+      submissionOptions = mkOption {
+        type = types.attrs;
+        default = {};
+        description = "Options for the submission config in master.cf";
+        example = { "smtpd_tls_security_level" = "encrypt";
```

Comment:
This looks much nicer! How about making that `example` the `default` choice of flags, though? Then people who want the default behavior can set `enableSubmission = true` and be done with it while everyone else still has the ability to specify custom flags.